### DISCUSSION TITLE
Feature extraction for whisper inference using recipe

### DISCUSSION THREAD

**Discussion by geoffvdr** [espnet/espnet#5053] [General]
Hi everyone,

I would like to evaluate the zero-shot performance of a whisper model on my own dataset, using the recipe ([https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/asr1/asr.sh](url)), something like this:
`./asr.sh --download_model openai/whisper-small --skip_data_prep false --skip_train true --skip_eval false`

How can I extract the exact same features as Whisper requires from my dataset?
Is there a way to use the model's `preprocessor_config` [https://huggingface.co/openai/whisper-small/blob/main/preprocessor_config.json](url) ?
I was thinking of extracting the filterbank features with the exact same parameters but the `fbank` feature extraction is not implemented in the recipe: [https://github.com/espnet/espnet/blob/master/egs2/TEMPLATE/asr1/asr.sh#L727-L730](url)

Thanks for your help.

Geoffroy

**Comment by sw005320** [comment]
We actually port the whisper frontend in espnet2/asr/frontend/whisper.py
So, we don't have to use espnet fbank.
You can check found some usages of whisper zeroshot in egs2/librispeech_100/asr1/README.md (or https://github.com/espnet/espnet/pull/4793)

@slSeanWU

**Comment by geoffvdr** [comment]
Hi @sw005320 , thank you for your reply.

It is still not clear to me how I can do zero-shot evaluation. egs2/librispeech_100/asr1/README.md seems to contain finetuned (3 epochs) models. [https://github.com/espnet/espnet/pull/4793](url) shows zero-shot results, however I still find it unclear how that is done.
 [egs2/librispeech_100/asr1/conf/tuning/train_asr_whisper_full.yaml](url) clarifies that the feature extraction is done within the whisper encoder, thanks for that. However It sounds like that `'train.yaml` is needed to do the inference? I find that confusing. It would be handy to be able to use `--download_model` and just start the recipe from stage 12.

**Comment by sw005320** [comment]
Yes, we specify the train config as well to perform inference.

**Comment by slSeanWU** [comment]
Hi @geoffvdr, the most straightforward way to do zero-shot inference (which is what I did) is to set
```
optim_conf:
    lr: 0.
max_epoch: 1
```
in the yaml config to perform a dry training run and export model weights to ESPnet format.

Don't forget to specify
```
model_conf:
    do_pad_trim: true
```
as whisper depends on this to perform well.

Let me know if this works!

**Reply by slSeanWU** [reply]
other than these yaml settings, you can use the same `run.sh` as you used for finetuning

**Reply by geoffvdr** [reply]
Got it, I will try that. Thanks a lot!
